File: trunk/esper/src/test/java/net/esper/eql/view/TestOutputConditionNull.java
package net.esper.eql.view;

import junit.framework.TestCase;
import net.esper.support.util.SupportUpdateListener;

public class TestOutputConditionNull extends TestCase
{

	private OutputConditionNull condition;	
    private SupportUpdateListener listener;
	private OutputCallback callback;

    public void setUp()
    { 
    	listener = new SupportUpdateListener();
    	
    	callback = new OutputCallback() {
		    		public void continueOutputProcessing(boolean doOutput, boolean forceUpdate)
		    		{
		    			listener.update(null, null);
		    		}
		    	};
		condition = new OutputConditionNull(callback);
    	
    }


    
    public void testUpdateCondition()
    {
    	// the callback should be made regardles of the update
    	condition.updateOutputCondition(true, 1,1);
    	assertTrue(listener.getAndClearIsInvoked());
    	condition.updateOutputCondition(true, 1,0);
    	assertTrue(listener.getAndClearIsInvoked());
    	condition.updateOutputCondition(true, 0,1);
    	assertTrue(listener.getAndClearIsInvoked());
    	condition.updateOutputCondition(true, 0,0);
    	assertTrue(listener.getAndClearIsInvoked());
    }
    

    

    public void testIncorrectUse()
	{
	    try
	    {
	    	condition = new OutputConditionNull(null);
	    	fail();
	    }
	    catch (NullPointerException ex)
	    {
	    	// Expected exception
	    }
	}



}

File: NEsper.Tests/eql/view/TestOutputConditionNull.cs
using System;

using net.esper.support.util;

using NUnit.Core;
using NUnit.Framework;

namespace net.esper.eql.view
{
	[TestFixture]
	public class TestOutputConditionNull 
	{
		private OutputConditionNull condition;
		private SupportUpdateListener listener;
		private OutputCallback callback;
		
		[SetUp]
		public virtual void  setUp()
		{
			listener = new SupportUpdateListener();
            callback = new OutputCallback(
                delegate(bool doOutput, bool forceUpdate)
                {
                    listener.Update(null, null);
                });
                
			condition = new OutputConditionNull(callback);
		}
		
		[Test]
		public virtual void  testUpdateCondition()
		{
			// the callback should be made regardles of the update
            condition.UpdateOutputCondition(1, 1);
			Assert.IsTrue(listener.GetAndClearIsInvoked());
            condition.UpdateOutputCondition(1, 0);
			Assert.IsTrue(listener.GetAndClearIsInvoked());
            condition.UpdateOutputCondition(0, 1);
			Assert.IsTrue(listener.GetAndClearIsInvoked());
            condition.UpdateOutputCondition(0, 0);
			Assert.IsTrue(listener.GetAndClearIsInvoked());
		}
		
		
		
		
		[Test]
		public virtual void  testIncorrectUse()
		{
			try
			{
				condition = new OutputConditionNull(null);
				Assert.Fail();
			}
			catch (System.NullReferenceException ex)
			{
				// Expected exception
			}
		}
	}
}
